src/overlays/actors/ovl_Obj_Warp2block/z_obj_warp2block.c

```diff
@@ -43,36 +42,305 @@ const ActorInit Obj_Warp2block_InitVars = {
+    this->dyna.actor.child->home.rot.z = temp;
+
+    Math_Vec3f_Copy(&tempVec, &this->dyna.actor.scale);
+    Math_Vec3f_Copy(&this->dyna.actor.scale, &this->dyna.actor.child->scale);
+    Math_Vec3f_Copy(&this->dyna.actor.child->scale, &tempVec);
+
+    Math_Vec3f_Copy(&tempVec, &this->dyna.actor.focus.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.focus.pos, &this->dyna.actor.child->focus.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.child->focus.pos, &tempVec);
+
+    tempRot = this->dyna.actor.focus.rot;
+    this->dyna.actor.focus.rot = this->dyna.actor.child->focus.rot;
+    this->dyna.actor.child->focus.rot = tempRot;
+
+    temp = this->dyna.actor.params & 0x7FFF;
+    this->dyna.actor.params = (this->dyna.actor.params & 0x8000) | (this->dyna.actor.child->params & 0x7FFF);
+    this->dyna.actor.child->params = (this->dyna.actor.child->params & 0x8000) | (temp  & 0x7FFF);
+
+    if (Math3D_Vec3fDistSq(&this->dyna.actor.world.pos, &this->dyna.actor.home.pos) < 0.01f) {
+        Flags_UnsetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
+    } else {
+    	Flags_SetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
+    }
+}
+
+s32 func_80BA2218(ObjWarp2block *this, GlobalContext *globalCtx) {
+	Player* player = PLAYER;
+
+    if (func_80BA1ECC(this, globalCtx)) {
+
+        if (player->stateFlags2 & 0x1000000) {
+            func_8010BD58(globalCtx, 1);
+            this->actionFunc_168 = func_80BA228C;
+        } else {
+            player->stateFlags2 |= 0x800000;
+        }
+    }
+
+    return 0;
+}
+
+
+s32 func_80BA228C(ObjWarp2block *this, GlobalContext *globalCtx) {
+    if (globalCtx->msgCtx.unk_E3EE == 4) {
+        this->actionFunc_168 = func_80BA2218;
+    }
+
+    if (globalCtx->msgCtx.unk_E3EC == 0xA) {
+        if (this->unk_172 == 0xFE) {
+            this->unk_16E = 0x6E;
+        } else {
+	        this->unk_16E = this->unk_16E - 1;
+	        if (this->unk_16E == 0) {
+	            return 1;
+	        }
+	    }
+    }
+    return 0;
+}
+
+
+s32 func_80BA2304(ObjWarp2block *this, GlobalContext *globalCtx) {
+    s32 ret;
+
+    ret = this->actionFunc_168(this, globalCtx);
+    this->unk_172 = globalCtx->msgCtx.unk_E3EC;
+
+    return ret;
+}
+
+
+void ObjWarp2block_Init(Actor* thisx, GlobalContext *globalCtx2) {
+	GlobalContext* globalCtx = globalCtx2;
+    ObjWarp2block* this = THIS;
+    CollisionHeader* collisionHeader;
+
+    collisionHeader = NULL;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.z = this->dyna.actor.shape.rot.z;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+
+    Actor_SetScale(&this->dyna.actor, sSpawnData[(this->dyna.actor.params >> 8) & 1].scale);
+    this->actionFunc_168 = func_80BA2218;
+    Actor_SetFocus(&this->dyna.actor, sSpawnData[(this->dyna.actor.params >> 8) & 1].focus);
+
+    if (((this->dyna.actor.params >> 0xF) & 1)) {
+        func_80BA24E8(this);
+        if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) != 0) {
```

Comment:
```suggestion
        if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F)) {
```
Flags_GetSwitch is treated as a boolean.